File: java_solutions/119.pascals-triangle-ii.java
/*
 * @lc app=leetcode id=119 lang=java
 *
 * [119] Pascal's Triangle II
 */

// semicolons : 1
class Solution {
    public List<Integer> getRow(int n) {
        return Stream.iterate(List.of(1), prev -> IntStream.range(0, prev.size() + 1).mapToObj(i -> (i == 0 || i == prev.size()) ? 1 : prev.get(i - 1) + prev.get(i)).toList()).skip(n).findFirst().orElse(List.of(1));
    }
}

File: kotlin_solutions/119.pascals-triangle-ii.kt
/*
 * @lc app=leetcode id=119 lang=kotlin
 *
 * [119] Pascal's Triangle II
 */

class Solution {fun getRow(rowIndex: Int): List<Int> = mutableListOf(1).apply { (1..rowIndex).forEach { i -> add((get(i - 1).toLong() * (rowIndex - i + 1) / i).toInt()) } }}